app/backend/src/couchers/servicers/communities.py

```diff
@@ -107,19 +104,25 @@ def GetCommunity(self, request, context):
 
     def ListCommunities(self, request, context):
         with session_scope() as session:
-            page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
-            next_node_id = int(request.page_token) if request.page_token else 0
+            page_size = request.page_size if request.page_size else MAX_PAGINATION_LENGTH
+            next_node_name = request.page_token if request.page_token else ""
+
             nodes = (
                 session.query(Node)
+                .select_from(Cluster)
```

Comment:
What does `select_from` do? We don't use it elsewhere, so it'd be good to not introduce it here.

Reply:
You probably want something along the lines of

```py3
session.query(Node, Cluster)
.join(Cluster, Cluster.parent_node_id == Node.id)
...
```